lib/kafka/consumer.rb

```diff
@@ -237,9 +241,11 @@ def each_message(min_bytes: 1, max_bytes: 10485760, max_wait_time: 1, automatica
             return if !@running
           end
 
-          # We've successfully processed a batch from the partition, so we can clear
-          # the pause.
-          pause_for(batch.topic, batch.partition).reset!
+          unless paused?(batch.topic, batch.partition)
+            # We've successfully processed a batch from the partition, so we can clear
+            # the pause state.
+            pause_for(batch.topic, batch.partition).reset!
+          end
```

Comment:
We might get to this line even if we _haven't_ processed any messages in the batch since the pause, hence the `unless`.